litmus-portal/graphql-server/pkg/file_handlers/file_handlers.go

```diff
@@ -28,56 +26,65 @@ var subscriberConfiguration = &types.SubscriberConfigurationVars{
 //FileHandler dynamically generates the manifest file and sends it as a response
 func FileHandler(w http.ResponseWriter, r *http.Request) {
 	var (
-		vars           = mux.Vars(r)
-		token          = vars["key"]
+		vars  = mux.Vars(r)
+		token = vars["key"]
+	)
+
+	response, statusCode, err := GetManifest(token)
+	if err != nil {
+		log.Print("error", err)
+		utils.WriteHeaders(&w, statusCode)
+	}
+
+	utils.WriteHeaders(&w, statusCode)
+	w.Write(response)
+}
+
+func GetManifest(token string) ([]byte, int, error) {
+	var (
 		portalEndpoint string
 	)
 
 	id, err := cluster.ClusterValidateJWT(token)
 	if err != nil {
-		log.Print("ERROR", err)
-		utils.WriteHeaders(&w, 404)
-		return
+		return nil, 404, err
 	}
 
 	reqCluster, err := database.GetCluster(id)
 	if err != nil {
-		log.Print("ERROR", err)
-		utils.WriteHeaders(&w, 500)
-		return
+		return nil, 500, err
 	}
 
 	if os.Getenv("PORTAL_SCOPE") == "cluster" {
 		portalEndpoint, err = k8s.GetPortalEndpoint()
 		if err != nil {
-			log.Print(err)
+			return nil, 500, err
 		}
 	} else if os.Getenv("PORTAL_SCOPE") == "namespace" {
 		portalEndpoint = os.Getenv("PORTAL_ENDPOINT")
 	}
 
 	subscriberConfiguration.GQLServerURI = portalEndpoint + "/query"
-
 	if !reqCluster.IsRegistered {
 		var respData []byte
 
 		if reqCluster.AgentScope == "cluster" {
 			respData, err = utils.ManifestParser(reqCluster, "manifests/cluster-subscriber.yml", subscriberConfiguration)
+			if err != nil {
+				return nil, 500, err
+			}
 		} else if reqCluster.AgentScope == "namespace" {
 			respData, err = utils.ManifestParser(reqCluster, "manifests/namespace-subscriber.yml", subscriberConfiguration)
+			if err != nil {
+				return nil, 500, err
+			}
 		} else {
 			log.Print("ERROR- AGENT SCOPE NOT SELECTED!")
 		}
 
-		if err != nil {
-			log.Print("ERROR", err)
-			utils.WriteHeaders(&w, 500)
-			return
-		}
-		utils.WriteHeaders(&w, 200)
-		w.Write(respData)
-		return
+		return respData, 200, nil
+	} else {
+		return []byte("Cluster is already registered"), 200, nil
```

Comment:
Should throw an error code not 200 OK

Reply:
you mean 404. Right?

Reply:
Could be 400 or 409 could also be good choices